authorize/grpc.go

```diff
@@ -23,6 +23,10 @@ func (a *Authorize) Check(ctx context.Context, in *envoy_service_auth_v3.CheckRe
 	ctx, span := trace.StartSpan(ctx, "authorize.grpc.Check")
 	defer span.End()
 
+	if err := a.WaitForInitialSync(ctx); err != nil {
```

Comment:
this was moved here because the initial sync can't begin until the outbound gRPC listener starts - i.e. the initial sync was waiting for the control plane to start, and the control plane was waiting for the initial sync to finish, so they were deadlocked.

Reply:
Do you mind adding adding a comment to that effect? (e.g. not the change, but why we are waiting)